Copy-edit 'Keyring support' section (#795)
pypa/twine: docs/keyring-support.rst

Keyring Support
===============

Instead of typing in your password every time you upload a distribution, Twine
allows storing a username and password securely using `keyring`_.
Keyring is installed with Twine but for some systems (Linux mainly) may
require `additional installation steps`_.

Once Twine is installed, use the ``keyring`` program to set a
username and password to use for each package index (repository) to
which you may upload.

For example, to set a username and password for PyPI:

.. code-block:: bash

    keyring set https://upload.pypi.org/legacy/ your-username

or

.. code-block:: bash

    python3 -m keyring set https://upload.pypi.org/legacy/ your-username

and enter the password when prompted.

For a different repository, replace the URL with the relevant repository

> URL. For example, for Test PyPI, use ``https://test.pypi.org/legacy/``.

Reviewer: ```suggestion
URL. For example, for `Test PyPI`_, use ``https://test.pypi.org/legacy/``.
```

Keyring Support
===============

Instead of typing in your password every time you upload a distribution, Twine
allows storing a username and password securely using `keyring`_.
Keyring is installed with Twine but for some systems (Linux mainly) may
require `additional installation steps`_.

Once Twine is installed, use the ``keyring`` program to set a
username and password to use for each package index (repository) to
which you may upload.

For example, to set a username and password for PyPI:

.. code-block:: bash

    keyring set https://upload.pypi.org/legacy/ your-username

or

.. code-block:: bash

>     python3 -m keyring set https://upload.pypi.org/legacy/ your-username

Reviewer 2: Do we really need to demo how runpy works in each example? Maybe we could just have one command invocation block in each place (w/o `python -m` but put a note somewhere that it's also possible).

Author: Maybe add a note before example:

```rst

.. note:: 

    Adding ``python3 -m`` before each command here also works.
```

Reviewer: I don't think the note is necessary; just the keyring command is fine.

and enter the password when prompted.

For a different repository, replace the URL with the relevant repository
URL. For example, for Test PyPI, use ``https://test.pypi.org/legacy/``.

> The next time you run ``twine``, it will prompt you for a username and will
> grab the appropriate password from the keyring.

Reviewer: ```suggestion
The next time you run ``twine``, it will prompt you for a username, and then
get the appropriate password from the keyring.
```

.. note::

>     If you are using Linux in a headless environment (such as on a
>     server) you'll need to do some additional steps to ensure that Keyring can
>     store secrets securely. See `Using Keyring on headless systems`_.

Reviewer 2: ```suggestion
   If you are using Linux in a headless environment (such as on a
   server) you'll need to do some additional steps to ensure that Keyring can
   store secrets securely. See `Using Keyring on headless systems`_.
```

Disabling Keyring
^^^^^^^^^^^^^^^^^

In most cases, simply not setting a password with ``keyring`` will allow Twine
to fall back to prompting for a password. In some cases, the presence of
Keyring will cause unexpected or undesirable prompts from the backing system.
In these cases, it may be desirable to disable Keyring altogether. To disable

> Keyring, simply invoke:

Reviewer: ```suggestion
Keyring, run:
```

.. code-block:: bash

    keyring --disable

or

.. code-block:: bash

    python -m keyring --disable

> That command will configure for the current user the "null" keyring,
> effectively disabling the functionality, and allowing Twine to prompt
> for passwords.
> 
> See `twine 338 <https://github.com/pypa/twine/issues/338>`_ for
> discussion and background.

Reviewer: The details here don't seem necessary.

```suggestion
See `Twine issue #338 <https://github.com/pypa/twine/issues/338>`_ for
discussion and background.
```

> .. _`keyring`: https://pypi.org/project/keyring/

Reviewer 2: nit: I believe one-word references don't really need backticks (needs checking)

Author: I've confirmed by this [guide](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages) and this [docs](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#toc-entry-36). Seems that you're right.

Author: However, I think it's better for us to unify these references with backticks. Adding backticks in one-word references won't affect the docs.

.. _`Using Keyring on headless systems`:

>     https://keyring.readthedocs.io/en/latest/#using-keyring-on-headless-linux-systems

Reviewer 2: If it's a sphinx project, we should use intershinx to be more resilient to the foreign docs restructurings.

Author: I got this error when I was trying to use `intersphinx`:

```text
Warning, treated as error:
~/twine/docs/index.rst:190:undefined label: #using-keyring-on-headless-linux-systems
```

But it works fine when using example from `sphinx` and `python docs`

Author: Only these can be used:

```shell
$ python -msphinx.ext.intersphinx https://keyring.readthedocs.io/en/latest/objects.inv
py:function
        keyring.delete_password                  index.html#keyring.delete_password
        keyring.get_credential                   index.html#keyring.get_credential
        keyring.get_keyring                      index.html#keyring.get_keyring
        keyring.get_password                     index.html#keyring.get_password
        keyring.set_keyring                      index.html#keyring.set_keyring
        keyring.set_password                     index.html#keyring.set_password
py:module
        keyring                                  index.html#module-keyring
std:doc
        history                                  History                                 : history.html
        index                                    Welcome to keyring documentation!       : index.html
std:label
        changes                                  History                                 : history.html#changes
        genindex                                 Index                                   : genindex.html
        modindex                                 Module Index                            : py-modindex.html
        py-modindex                              Python Module Index                     : py-modindex.html
        search                                   Search Page                             : search.html
```

Reviewer 3: I think it would have to be some kind of 

```rst
:std:doc`index`
```
Usage but I think that will conflict with other doc references :/ I don't think intersphinx has great namespacing unless the docs just don't mention it

Author: @sigmavirus24 Thanks for your suggestion! I've tested it and seems that it will only point to the index page of the docs. It won't point to the section "using-keyring-on-headless-linux-systems".
